docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,229 @@
+* New commands will be added using flags (e.g., `--abc`) or environment variables (e.g., `ILAB_ABC`) to avoid any configuration changes. 
+  This approach ensures configuration compatibility even if the commands are deprecated later.
+
+### 2.2 Embedding Ingestion Pipeline Command
+The proposal is to add a `process`  command to the `data` command group, with an explicit `--rag` flag to trigger
+the executions:
+```
+ilab data process --rag /path/to/docs/folder
+```
+
+The rationale behind this choice is that the `data process` command can support future workflows, making its
+introduction an investment to anticipate other needs.
+
+Since the RAG behavior is the only functionality of this new command, executions without the `--rag` option will result
+in no output for now.
+
+#### Command Purpose
+Generate the embeddings from the documents at */path/to/docs/folder* folder and store them in the
+configured vector database. These embeddings are intended to be used as augmented context in a RAG-based chat pipeline.
+
+#### Assumptions
+The provided documents must be in JSON format according to the InstructLab schema: this is the schema generated
+when transforming knowledge documents with the `ilab data generate` command (see 
+[Getting Started with Knowledge Contributions][1]).
+
+To simplify the execution of the transformation step, we introduce a `--transform` option which also includes the documents
+transformation, leveraging on the `instructlab-sdg` modules. 
+
+### Why We Need It
+This command streamlines the `ilab data generate` pipeline and eliminates the requirement to define a `qna` document,
+which typically includes:
+* A minimum of 5×3 question-and-answer pairs.
+* Reference documents stored in Git.
+
+The goal is not to generate training data for InstructLab-trained models but to utilize the documents for RAG 
+workflows with pre-tuned models.
+
+#### Supported Databases
+The command may support various vector database types. A default configuration will align with the selected 
+InstructLab technology stack.
+
+#### Usage
+The generated embeddings can later be retrieved from a vector database and converted to text, enriching the 
+context for RAG-based chat pipelines.
+
+#### Defining Command Options
+To maintain compatibility and simplicity, no new configurations will be introduced for this command. Instead,
+the settings will be defined using the following hierarchy (options higher in the list overriding those below):
+* CLI flags (e.g., `--rag`).
+* Environment variables following a consistent naming convention, such as `ILAB_<UPPERCASE_ARGUMENT_NAME>`.
+* Default values, for all the applicable use cases.
+
+For example, the `vectordb-uri` argument can be implemented using the `click` module like this:
+```py
+@click.option(
+    "--vectordb-uri",
+    default='rag-output.db',
+    envvar="ILAB_VECTORDB_URI",
+)
+```
+
+### 2.3 Embedding Ingestion Pipeline Options
```

Comment:
I wonder if instead of wiring special support for all this into ilab, we should just consider generating and inserting vector embeddings stages in a data generation pipeline? We're moving to a model where users can supply their own custom pipeline and create their own custom pipeline blocks. So, we should ship (or the user could define) a RAG pipeline that handles turning the data samples into embeddings and storing them in a vector database all as part of our existing pipeline flows, without any code changes in SDG itself?

Reply:
Is there any design document that you can share about this initiative?